tools/importer-rest-api-specs/components/schema/builder.go

```diff
@@ -269,7 +268,7 @@ func (b Builder) schemaFromTopLevelModel(input resourcemanager.TerraformResource
 	if !ok {
 		return nil, fmt.Errorf("couldn't find Resource ID named %q", input.ResourceIdName)
 	}
-	fieldsWithinResourceId, mappings, err := b.identifyTopLevelFieldsWithinResourceID(resourceId, mappings, input.DisplayName, logger.Named("TopLevelFields ResourceID"))
+	fieldsWithinResourceId, mappings, err := b.identifyTopLevelFieldsWithinResourceID(resourceId, mappings, &input, logger.Named("TopLevelFields ResourceID"))
```

Comment:
After investigating, I don't agree with this.

The overrides that we specify are tied to a specific resource, in a specific version, in a specific service. Pulling them out of the `resourcemanager.TerraformResourceDetails` struct means we lose the information required to decide on what resource, in which version and service to apply the override. 

This update would require us to either nest this secondary struct containing the overrides within a resource, version, service struct or add metadata to the secondary struct containing resource, version, service to be able to determine to which resource/version/service an override applies to. I don't believe the overhead and complexity that this adds is worth the benefit of it remaining scoped to `importer-rest-api-specs`.

I'm open to revisiting this when we discuss boundaries/separation of concerns for refactoring and making Pandora more manageable.

Reply:
> The overrides that we specify are tied to a specific resource, in a specific version, in a specific service. Pulling them out of the `resourcemanager.TerraformResourceDetails` struct means we lose the information required to decide on what resource, in which version and service to apply the override.

Given the limitation (that I don't expect we'd change) that each Terraform Resource is tied to one specific API Version - therefore we’ve got the Service / API Version for each Resource ([available in these types](https://github.com/hashicorp/pandora/blob/06969ad36ac0061859ad35a3da0f6ad69d11aaa8/tools/sdk/config/definitions/interface.go#L8-L41) which come from the config files [where the `override` information will be defined]).

Since we’re iterating over each resource, we’ve got access to the Service, API Version and Resource (Label) in question - so I think it should be possible to [pull that out here](https://github.com/hashicorp/pandora/blob/06969ad36ac0061859ad35a3da0f6ad69d11aaa8/tools/importer-rest-api-specs/pipeline/run_importer.go#L99-L102), where we build up the Terraform Resource, Tests and Example Usage - for example:

```diff
diff --git a/tools/importer-rest-api-specs/pipeline/run_importer.go b/tools/importer-rest-api-specs/pipeline/run_importer.go
index 48dd2a8719..5061ce00c1 100644
--- a/tools/importer-rest-api-specs/pipeline/run_importer.go
+++ b/tools/importer-rest-api-specs/pipeline/run_importer.go
@@ -120,21 +120,40 @@ func runImportForService(input RunInput, serviceName string, apiVersionsForServi
                        }
                }

+               type resourceBuildInfo struct {
+                       // TODO: data
+               }
+
+               var buildInfoForResource *resourceBuildInfo
+               if api[0].TerraformServiceDefinition != nil {
+                       for version, versionDetails := range api[0].TerraformServiceDefinition.ApiVersions {
+                               for pkgName, pkgDetails := range versionDetails.Packages {
+                                       resDetails, ok := pkgDetails.Definitions["resource_label"]
+                                       if ok {
+                                               buildInfoForResource = &resourceBuildInfo{
+                                                       Example: resDetails.Overrides.Example, // picking a random field from the override config for this Resource
+                                               }
+                                               break
+                                       }
+                               }
+                       }
+               }
+
                versionLogger.Trace("generating Terraform Details")
                var err error
-               dataForApiVersion, err = task.generateTerraformDetails(dataForApiVersion, versionLogger.Named("TerraformDetails"))
+               dataForApiVersion, err = task.generateTerraformDetails(dataForApiVersion, buildInfoForResource, versionLogger.Named("TerraformDetails"))
                if err != nil {
                        return fmt.Errorf(fmt.Sprintf("generating Terraform Details for Service %q / Version %q: %+v", serviceName, apiVersion, err))
                }

                versionLogger.Trace("generating Terraform Tests")
-               dataForApiVersion, err = task.generateTerraformTests(dataForApiVersion, input.ProviderPrefix, versionLogger.Named("TerraformTests"))
+               dataForApiVersion, err = task.generateTerraformTests(dataForApiVersion, input.ProviderPrefix, buildInfoForResource, versionLogger.Named("TerraformTests"))
                if err != nil {
                        return fmt.Errorf(fmt.Sprintf("generating Terraform Tests for Service %q / Version %q: %+v", serviceName, apiVersion, err))
                }

                versionLogger.Trace("Generating Example Usage from the Terraform Tests")
-               dataForApiVersion, err = task.generateTerraformExampleUsage(dataForApiVersion, input.ProviderPrefix, versionLogger.Named("TerraformExampleUsage"))
+               dataForApiVersion, err = task.generateTerraformExampleUsage(dataForApiVersion, input.ProviderPrefix, buildInfoForResource, versionLogger.Named("TerraformExampleUsage"))
                if err != nil {
                        return fmt.Errorf(fmt.Sprintf("generating Terraform Example Usage for Service %q / Version %q: %+v", serviceName, apiVersion, err))
                }
```

Whilst the `TerraformResourceDetails` type is doing a little more than it should today - adding more information into this (and particularly persisting it to disk/the Data API) is going to dramatically increase the complexity down the line.

As such I think we want to make a concerted effort to prevent this information from leaking outside of the `importer-rest-api-specs` tool for now - meaning that we need to ensure these fields aren’t included in `TerraformResourceDetails`.

Ultimately this ensures that the `importer-rest-api-specs` tool is the only thing concerned with the overrides - the `TerraformResourceDetails` that's written to the API Definitions contains just the overridden result (e.g. the Terraform Resource with any overrides applied) meaning that other tools (Data API, Go SDK Generator etc) are unaware that any overrides have been applied - meaning they're focused on one thing (and doing one thing well).